builder/format.go

```diff
@@ -39,16 +39,21 @@ func (b *Builder) UpdateFormatVersion(version string) error {
 		return errors.Wrap(err, "Failed to read builder.conf")
 	}
 
-	builderEdit := re.ReplaceAllString(string(builderData), newver)
+	var builderEdit string
+	if !UseNewConfig {
```

Comment:
The flow of this whole function is really weird now, with a bunch of stuff being created and never used for certain paths, and back-and-forth between strings and bytes. 

I think it can be simplified to:
```go
func (b *Builder) UpdateFormatVersion(version string) error {
	b.Config.Swupd.Format = version

	if UsenewConfig {
		var config string
		if config, err = GetConfigPath(config); err != nil {
			return err
		}

		var mc MixConfig
		if err = mc.LoadConfig(config); err != nil {
			return err
		}

		return mc.SetProperty(config, "Swupd.FORMAT", version)
	}

	builderData, err := ioutil.ReadFile(b.BuildConf)
	if err != nil {
		return errors.Wrap(err, "Failed to read builder.conf")
	}

	var re = regexp.MustCompile(`(FORMAT=)[0-9]+`)
	newver := []byte("${1}" + b.Config.Swupd.Format)
	builderData = re.ReplaceAll(builderData, newver)
	
	if err = ioutil.WriteFile(b.BuildConf, builderData, 0644); err != nil {
		return errors.Wrap(err, "Failed to write new builder.conf")
	}

	return nil
}
```

Notes:
- I changed your regex from `(FORMAT=)[0-9]*` to `(FORMAT=)[0-9]+`. Feel free to change it back if you want, but I didn't think we supported an empty format value (it's listed as `required:"true"`).
- @rchiossi -- the above usage illustrates that it'd probably be good to have a version of `SetProperty` that operates on the **current config file from which the `MixConifg` object was built**, so that we could just call `b.Config.SetProperty("Swupd.FORMAT", version")`